src/abstractdataframe/selection.jl

```diff
@@ -254,17 +257,28 @@ function normalize_selection(idx::AbstractIndex,
                 end
             end
     end
+
     if lls isa AbstractString
-        r = Symbol(lls)
+        combine_target_col = Symbol(lls)
     elseif lls isa AbstractVector{<:AbstractString}
-        r = Symbol.(lls)
+        combine_target_col = Symbol.(lls)
     else
-        r = lls
+        combine_target_col = lls
     end
-    if r isa AbstractVector{Symbol}
-        allunique(r) || throw(ArgumentError("target column names must be unique"))
+
+    if combine_target_col isa AbstractVector{Symbol}
+        allunique(combine_target_col) || throw(ArgumentError("target column names must be unique"))
     end
-    return (wanttable ? AsTable(c) : c) => first(last(sel)) => r
+
+    combine_src = if wanttable 
+        AsTable(c) 
+    else 
+        (length(c) == 1 ? only(c) : c)
+    end
```

Comment:
```suggestion
    combine_src = wanttable ? AsTable(c) : (length(c) == 1 ? only(c) : c)
```

Reply:
In terms of readability, I think clarity is more important than conciseness here.

Reply:
In my opinion your version is less readable (of course this is subjective). Also normally users do not expect `if-else` to return a value (it does in Julia, but not e.g. in Python). If you insist on your version then I would do:
```
    if wanttable 
        combine_src = AsTable(c) 
    else 
        combine_src = (length(c) == 1 ? only(c) : c)
    end
```